tests/pkg/testing/testing.go

```diff
@@ -18,6 +18,48 @@ func NewT(t *testing.T) *T {
 	}
 }
 
+type Logger interface {
+	Log(args ...any)
+	Logf(format string, args ...any)
+	Errorf(format string, args ...interface{})
+	FailNow()
+}
+
+type LoggerWithOnlyOnce struct {
```

Comment:
How does it sound "OnceLogger" or "LoggerOnce" to you? Or something similar? 
And we could also add a short godoc about the idea of this logger and recommendations to use it together with the retry/async asserts to reduce the logs